src/main/java/games/strategy/util/memento/PropertyBagMemento.java

```diff
@@ -0,0 +1,103 @@
+package games.strategy.util.memento;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Implementation of {@link Memento} that represents the originator state as a property bag.
```

Comment:
> Replacing the words 'property bag' with a definition for them would likely help.

I'll cover this when I review the Javadocs, as mentioned above.

> Also would be interesting to have noted any limitations of what can be property bagged and what cannot.

From this class's perspective, there are no limitations.  When _persisting_ an instance of this class, the underlying persistence layer may impose limitations on what value types may be persisted.  I'm not sure if this class should be discussing persistence concerns.